src/lib.rs

```diff
@@ -13,3 +12,154 @@ pub fn insert_bootloader_input(
 ) {
     exec_scopes.insert_value(BOOTLOADER_INPUT, bootloader_input);
 }
+
+#[macro_export]
+macro_rules! vm {
+    () => {
+        VirtualMachine::new(false) // Default to false if no argument is provided
+    };
+    ($trace_enabled:expr) => {
+        VirtualMachine::new($trace_enabled)
+    };
+}
+
+#[macro_export]
+macro_rules! add_segments {
+    ($vm:expr, $n:expr) => {
+        for _ in 0..$n {
+            $vm.add_memory_segment();
+        }
+    };
+}
+#[macro_export]
+macro_rules! ids_data_old {
+    ( $( $name: expr ),* ) => {
+        {
+            #[allow(clippy::useless_vec)]
+            let ids_names = vec![$( $name ),*];
+            let references = $crate::references!(ids_names.len() as i32);
+            let mut ids_data = HashMap::<String, cairo_vm::hint_processor::hint_processor_definition::HintReference>::new();
+            for (i, name) in ids_names.iter().enumerate() {
+                ids_data.insert(name.to_string(), references.get(&i).unwrap().clone());
+            }
+            ids_data
+        }
+    };
+}
+#[macro_export]
+macro_rules! ids_data {
```

Comment:
Same, can be removed I guess. Most macros in this file are not used and are just copies of what's in `cairo-vm`, clean this up.